src/platform/packages/private/kbn-grid-layout/grid/grid_panel/use_panel_grid_data.tsx

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { useEffect, useMemo, useRef } from 'react';
+import { BehaviorSubject, distinctUntilChanged, filter, map } from 'rxjs';
+
+import type { OrderedLayout } from '../types';
+import { useGridLayoutContext } from '../use_grid_layout_context';
+import { isGridDataEqual } from '../utils/equality_checks';
+import type { GridPanelData } from './types';
+
+export const useGridPanelState = ({
+  panelId,
+}: {
+  panelId: string;
+}): BehaviorSubject<(GridPanelData & { sectionId: string }) | undefined> => {
+  const { gridLayoutStateManager } = useGridLayoutContext();
+  const cleanupCallback = useRef<null | (() => void)>();
+
+  const panel$ = useMemo(() => {
+    const panelSubject = new BehaviorSubject(
+      getPanelState(gridLayoutStateManager.gridLayout$.getValue(), panelId)
+    );
+
+    const subscription = gridLayoutStateManager.gridLayout$
+      .pipe(
+        map((layout) => getPanelState(layout, panelId)),
+        // filter out undefined panels
+        filter((panel) => Boolean(panel)),
+        distinctUntilChanged(
+          (panelA, panelB) =>
+            isGridDataEqual(panelA, panelB) && panelA!.sectionId === panelB!.sectionId
+        )
+      )
+      .subscribe((panel) => {
+        panelSubject.next(panel as GridPanelData & { sectionId: string }); // undefined panels were filtered out
```

Comment:
```suggestion
        filter(nonNullable),
        distinctUntilChanged(
          (panelA, panelB) =>
            isGridDataEqual(panelA, panelB) && panelA!.sectionId === panelB!.sectionId
        )
      )
      .subscribe((panel) => {
        panelSubject.next(panel); 
```

to avoid casting, you could define a helper:

export function nonNullable<T>(value: T): value is NonNullable<T> {
  return value !== null && value !== undefined;
}

and then filter undefined values via it :)

Reply:
Nice - done in 7cc7606a503ab3ed70092fe6a8aab6172a9b134f